core/src/integration-test/java/com/scalar/db/storage/jdbc/JdbcAdminImportTestUtils.java

```diff
@@ -0,0 +1,442 @@
+package com.scalar.db.storage.jdbc;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.config.DatabaseConfig;
+import com.scalar.db.io.DataType;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.apache.commons.dbcp2.BasicDataSource;
+
+public class JdbcAdminImportTestUtils {
+  static final List<String> UNSUPPORTED_DATA_TYPES_MYSQL =
+      Arrays.asList(
+          "BIGINT UNSIGNED",
+          "BIT(8)",
+          "DATE",
+          "DATETIME",
+          "DECIMAL(8,2)",
+          "ENUM('a','b')",
+          "SET('a','b')",
+          "GEOMETRY",
+          "JSON", // we remove this for MariaDB because it is an alias of a supported type, LONGTEXT
+          "NUMERIC",
+          "TIME",
+          "TIMESTAMP",
+          "YEAR");
+  static final List<String> UNSUPPORTED_DATA_TYPES_PGSQL =
+      Arrays.asList(
+          "bigserial",
+          "bit(8)",
+          "bit varying(8)",
+          "box",
+          "cidr",
+          "circle",
+          "date",
+          "inet",
+          "interval",
+          "json",
+          "jsonb",
+          "line",
+          "lseg",
+          "macaddr",
+          "macaddr8",
+          "money",
+          "numeric(8,2)",
+          "path",
+          "pg_lsn",
+          "pg_snapshot", // after v14
+          "point",
+          "polygon",
+          "smallserial",
+          "time",
+          "time with time zone",
+          "timestamp",
+          "timestamp with time zone",
+          "tsquery",
+          "tsvector",
+          "txid_snapshot",
+          "uuid",
+          "xml");
+  static final List<String> UNSUPPORTED_DATA_TYPES_ORACLE =
+      Arrays.asList(
+          "BFILE",
+          "DATE",
+          "FLOAT(54)",
+          "INT",
+          "INTERVAL YEAR(3) TO MONTH",
+          "INTERVAL DAY(2) TO SECOND",
+          "JSON",
+          "NUMBER(16,0)",
+          "ROWID",
+          "TIMESTAMP",
+          "TIMESTAMP WITH TIME ZONE",
+          "TIMESTAMP WITH LOCAL TIME ZONE",
+          "UROWID");
+  static final List<String> UNSUPPORTED_DATA_TYPES_MSSQL =
+      Arrays.asList(
+          "date",
+          "datetime",
+          "datetime2",
+          "datetimeoffset",
+          "decimal(8,2)",
+          "hierarchyid",
+          "money",
+          "numeric(8,2)",
+          "rowversion",
```

Comment:
Do you mean why we do not have to add the timestamp data type here? If yes, that is because there is no such data type in the reference. I'm not really sure, but I don't think `rowversion` is an alternative for `timestamp` since `datetime` and `datetime2` are mapped to JDBC `TIMESTAMP`. Note that the document you pointed out also says that:

> The Transact-SQL timestamp data type is different from the timestamp data type defined in the ISO standard.

Anyway, we don't have to have `timestamp` data type here, I think.

Reply:
I see. Thanks for the explanation!